File: sorting_algorithms.py
def bubble_sort(arr):
    """
    Implementation of Bubble Sort algorithm
    """
    # Make a copy of the list to avoid modifying the original
    result = arr.copy()
    
    # TODO: Implement Bubble Sort Algorithm
    
    return result

def selection_sort(arr):
    """
    Implementation of Selection Sort algorithm
    """
    # Make a copy of the list to avoid modifying the original
    result = arr.copy()
    
    # TODO: Implement Selection Sort Algorithm
    
    return result

def insertion_sort(arr):
    """
    Implementation of Insertion Sort algorithm
    """
    # Make a copy of the list to avoid modifying the original
    result = arr.copy()
    
    # TODO: Implement Insertion Sort Algorithm
    
    return result

def merge_sort(arr):
    """
    Implementation of Merge Sort algorithm
    """
    # Make a copy of the list to avoid modifying the original
    result = arr.copy()
    
    # TODO: Implement Merge Sort Algorithm
    
    return result

def quick_sort(arr):
    """
    Implementation of Quick Sort algorithm
    """
    # Make a copy of the list to avoid modifying the original
    result = arr.copy()
    
    # TODO: Implement Quick Sort Algorithm
    
    return result

def print_array(arr):
    """
    Helper function to print arrays
    """
    print(arr)

def main():
    # Read input from file
    try:
        with open('input.txt', 'r') as file:
            line = file.readline().strip()
            # Convert comma-separated string to list of integers
            original_array = [int(x.strip()) for x in line.split(',')]
    except Exception as e:
        print(f"Error reading input file: {str(e)}")
        return
    
    # Display original array
    print("Original array:")
    print_array(original_array)
    
    # Test Bubble Sort
    print("\nBubble Sort:")
    bubble_sorted = bubble_sort(original_array)
    print_array(bubble_sorted)
    
    # Test Selection Sort
    print("\nSelection Sort:")
    selection_sorted = selection_sort(original_array)
    print_array(selection_sorted)
    
    # Test Insertion Sort
    print("\nInsertion Sort:")
    insertion_sorted = insertion_sort(original_array)
    print_array(insertion_sorted)
    
    # Test Merge Sort
    print("\nMerge Sort:")
    merge_sorted = merge_sort(original_array)
    print_array(merge_sorted)
    
    # Test Quick Sort
    print("\nQuick Sort:")
    quick_sorted = quick_sort(original_array)
    print_array(quick_sorted)
    
    # Display original array again for comparison
    print("\nOriginal array (unchanged):")
    print_array(original_array)

if __name__ == "__main__":
    main()

File: SortingAlgorithms.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Arrays;

//Test Commit and push

public class SortingAlgorithms {
    
    // Bubble Sort Implementation
    public static int[] bubbleSort(int[] array) {
        // Make a copy of the array to avoid modifying the original
        int[] result = Arrays.copyOf(array, array.length);
        
        // TODO: Implement Bubble Sort Algorithm
        
        return result;
    }
    
    // Selection Sort Implementation
    public static int[] selectionSort(int[] array) {
        // Make a copy of the array to avoid modifying the original
        int[] result = Arrays.copyOf(array, array.length);
        
        // TODO: Implement Selection Sort Algorithm
        
        return result;
    }
    
    // Insertion Sort Implementation
    public static int[] insertionSort(int[] array) {
        // Make a copy of the array to avoid modifying the original
        int[] result = Arrays.copyOf(array, array.length);
        
        // TODO: Implement Insertion Sort Algorithm
        
        return result;
    }
    
    // Merge Sort Implementation
    public static int[] mergeSort(int[] array) {
        // Make a copy of the array to avoid modifying the original
        int[] result = Arrays.copyOf(array, array.length);
        
        // TODO: Implement Merge Sort Algorithm
        
        return result;
    }
    
    // Quick Sort Implementation
    public static int[] quickSort(int[] array) {
        // Make a copy of the array to avoid modifying the original
        int[] result = Arrays.copyOf(array, array.length);
        
        // TODO: Implement Quick Sort Algorithm
        
        return result;
    }
    
    // Helper method to print arrays
    public static void printArray(int[] array) {
        System.out.print("[");
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i]);
            if (i < array.length - 1) {
                System.out.print(", ");
            }
        }
        System.out.println("]");
    }
    
    // Main method to test sorting algorithms
    public static void main(String[] args) {
        int[] array = null;
        
        // Read input from file
        try (BufferedReader br = new BufferedReader(new FileReader("input.txt"))) {
            String line = br.readLine();
            String[] strNumbers = line.split(",");
            array = new int[strNumbers.length];
            
            for (int i = 0; i < strNumbers.length; i++) {
                array[i] = Integer.parseInt(strNumbers[i].trim());
            }
        } catch (IOException e) {
            System.err.println("Error reading input file: " + e.getMessage());
            return;
        }
        
        // Display original array
        System.out.println("Original array:");
        printArray(array);
        
        // Test Bubble Sort
        System.out.println("\nBubble Sort:");
        int[] bubbleSorted = bubbleSort(array);
        printArray(bubbleSorted);
        
        // Test Selection Sort
        System.out.println("\nSelection Sort:");
        int[] selectionSorted = selectionSort(array);
        printArray(selectionSorted);
        
        // Test Insertion Sort
        System.out.println("\nInsertion Sort:");
        int[] insertionSorted = insertionSort(array);
        printArray(insertionSorted);
        
        // Test Merge Sort
        System.out.println("\nMerge Sort:");
        int[] mergeSorted = mergeSort(array);
        printArray(mergeSorted);
        
        // Test Quick Sort
        System.out.println("\nQuick Sort:");
        int[] quickSorted = quickSort(array);
        printArray(quickSorted);
        
        // Display original array again for comparison
        System.out.println("\nOriginal array (unchanged):");
        printArray(array);
    }
}
